src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -1043,6 +1050,77 @@ internal static List<AzureSqlSkuCategory> GetEligibleSkuCategories(string target
             return eligibleSkuCategories;
         }
 
+        /// <summary>
+        /// Request handler for the certifica migration operation
+        /// </summary>
+        /// <param name="parameters">Parameters for the operation, as register during the type definition</param>
+        /// <param name="requestContext">Context provided by the framework</param>
+        /// <returns></returns>
+        internal async Task HandleCertificateMigrationRequest(
+          CertificateMigrationParams parameters,
+          RequestContext<CertificateMigrationResult> requestContext)
+        {
+            var result = new CertificateMigrationResult();
+
+            var credentials = new StaticTokenCredential(parameters.AccessToken); //New token provided, will change to shared ADS cache later.
+
+            // Reuse the tde migration client
+            var tdeMigrationClient = new TdeMigration(
+                   parameters.SourceSqlConnectionString,
+                   parameters.TargetSubscriptionId,
+                   parameters.TargetResourceGroupName,
+                   parameters.TargetManagedInstanceName,
+                   parameters.NetworkSharePath,
+                   parameters.NetworkShareDomain,
+                   parameters.NetworkShareUserName,
+                   parameters.NetworkSharePassword,
+                   credentials
+                   );
+
+            foreach (var dbName in parameters.EncryptedDatabases)
```

Comment:
I remember one of Steven's demo talked about parallelizing migrating for dbs. just curious what the outcome of that was?

Reply:
The operation that consumes most of the time, is getting an MI instance from the ARM client. We are getting that one once and reusing it later. 
Parallelizing would be for the next phase when we will move these steps inside the nuget package and try to review how retries are handled by the ARM client or whether they can be configured, so we don't get 429 from azure.